Fixes fitness bug that was crashing for Parajke stat on dDEM obs.

mjbrodzik opened the pull request:
Parajke processing assumes 3 dimensions in obs data, which works
fine for MODIS (time, rows, cols) but was causing an error in the
call to sub2ind due to incorrect indexing when dDEM obs (rows,
cols) were being compared.  In this case, have temporarily forced
a degenerate 3rd dimension of 1 (time step) so that indexing is
done correctly.

Note that call to squeeze for mod and obs at end will squeeze all
dimensions of size 1, so if there is ever a case where a vector
is input, the resulting obs and mod arrays may be squeezed too far.

Commit:
Fixes fitness bug that was crashing for Parajke stat on dDEM obs.

Parajke processing assumes 3 dimensions in obs data, which works
fine for MODIS (time, rows, cols) but was causing an error in the
call to sub2ind due to incorrect indexing when dDEM obs (rows,
cols) were being compared.  In this case, have temporarily forced
a degenerate 3rd dimension of 1 (time step) so that indexing is
done correctly.

Note that call to squeeze for mod and obs at end will squeeze all
dimensions of size 1, so if there is ever a case where a vector
is input, the resulting obs and mod arrays may be squeezed too far.

thomasmosier reviewed (changes requested):
The first new snippet of code looks good and addresses the issue that you encountered. The second code snippet (lines 391-397) does no harm but is unnecessary because obs and mod are not called again. I therefore recommend removing the squeeze and simplifying the first code snippet (e.g. removing the "orig2dim" variable + else clause).

mjbrodzik commented:
ok thanks for the fast look.  more tomorrow.

Sent from my iPhone, but not with thumbs.

On Dec 16, 2019, at 17:34, Thomas Mosier <notifications@github.com> wrote:

﻿

@thomasmosier requested changes on this pull request.

The first new snippet of code looks good and addresses the issue that you encountered. The second code snippet (lines 391-397) does no harm but is unnecessary because obs and mod are not called again. I therefore recommend removing the squeeze and simplifying the first code snippet (e.g. removing the "orig2dim" variable + else clause).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/thomasmosier/shared_subroutines_Matlab/pull/4?email_source=notifications&email_token=ABGIV2B53JQTGCU2KCWSGHDQZANC5A5CNFSM4J3TOPGKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPMIMDA#pullrequestreview-332957196>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGIV2AWWUNPYVSWJBNH3RLQZANC5ANCNFSM4J3TOPGA>.

Commit:
Removed the unnecessary reset to original shape.

I was worried that Matlab passes by reference and that I'd be
reshaping the mod and obs arrays in the calling routine.
Apparently it uses "copy on
write" (https://www.mathworks.com/matlabcentral/answers/450892-passing-by-reference-vs-value)
so this reshaping back to original is unnecessary.
